maven-core/src/test/java/org/apache/maven/plugin/PluginParameterExpressionEvaluatorV4Test.java

```diff
@@ -0,0 +1,104 @@
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.maven.AbstractCoreMavenComponentTestCase;
+import org.apache.maven.api.Session;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.ArtifactUtils;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.DefaultMavenExecutionRequest;
+import org.apache.maven.execution.DefaultMavenExecutionResult;
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.internal.impl.DefaultSession;
+import org.apache.maven.model.Build;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.plugin.descriptor.MojoDescriptor;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.project.DuplicateProjectException;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.repository.RepositorySystem;
+import org.codehaus.plexus.MutablePlexusContainer;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluator;
+import org.codehaus.plexus.util.dag.CycleDetectedException;
+import org.junit.jupiter.api.Test;
+
+import static org.codehaus.plexus.testing.PlexusExtension.getTestFile;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class PluginParameterExpressionEvaluatorV4Test
+    extends AbstractCoreMavenComponentTestCase
+{
+
+    @Test
+    public void testUri()
+            throws Exception
+    {
+        Path path = Paths.get( "" ).toAbsolutePath();
+
+        MavenSession mavenSession = createMavenSession( null );
+        mavenSession.getRequest().setBaseDirectory( path.toFile() );
+        mavenSession.getRequest().setMultiModuleProjectDirectory( path.toFile() );
+
+        Object result = new PluginParameterExpressionEvaluatorV4( mavenSession.getSession(), null )
+                .evaluate( "${session.multiModuleProjectDirectory.uri}" );
+        assertEquals( path.toUri(), result );
+    }
+
+    @Test
+    public void testPath()
+            throws Exception
+    {
+        Path path = Paths.get( "" ).toAbsolutePath();
+
+        MavenSession mavenSession = createMavenSession( null );
+        mavenSession.getRequest().setBaseDirectory( path.toFile() );
+        mavenSession.getRequest().setMultiModuleProjectDirectory( path.toFile() );
+
+        Object result = new PluginParameterExpressionEvaluatorV4( mavenSession.getSession(), null )
+                .evaluate( "${session.multiModuleProjectDirectory/target}" );
```

Comment:
I am a bit confused where this is documented with the `/`.

Reply:
I'm not sure it's documented anywhere, but the code is there, so I added a small test for both.

Reply:
The code is there:
https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluator.java#L209-L216

It has been added 13 years ago by https://github.com/apache/maven/commit/f601e9c0972810aefa579c9501c25e5bb88e3a36